### ISSUE TITLE
[Bug] ClubChatRoom 삭제 시 TransientObjectException 발생

### ISSUE THREAD

**Issue by Developer-Groo** [Developer-Groo/Mobble-Server#125]
## ❗ 버그 설명
Club ↔ ClubChatRoom 1:1 연관관계에서 flush 시점에 TransientObjectException 예외가 발생합니다.

## 🔁 재현 방법
1. 클럽 생성 API 호출 (예: POST /api/clubs)
2. 서비스 내부에서 ClubChatRoom 생성 직후(아직 save 전) 클럽/채팅방 삭제 API 호출 시 club_chat_room 삭제 전에 역방향을 끊지 않음
3. 이후 다른 리포지토리 호출/벌크 쿼리 실행으로 auto-flush 트리거

## 🖼️ 스크린샷 또는 로그
~~~ bash
Caused by: org.hibernate.TransientObjectException: persistent instance references an unsaved transient instance of 'com.mobble.mobbleserver.domain.chat.clubChatRoom.entity.ClubChatRoom' (save the transient instance before flushing)
~~~

## ✅ 예상되는 해결 방향 (선택)
- 연관 관리 메서드 도입(소유측 주도)
    - ClubChatRoom.attachTo(Club) : save 후 역방향만 동기화
    - ClubChatRoom.detach() : 파라미터 없이 양쪽 참조 해제(역방향 null → 소유측 null)
    - Club.setClubChatRoomInternal(ClubChatRoom) : null 허용(연관 해제 목적)
- 생성/삭제 순서 표준화
    - 생성: ChatRoom → ChatRoomParticipant → ClubChatRoom(save) → clubChatRoom.attachTo(club)
    - 삭제: clubChatRoom.detach() 먼저 → 참가자/메시지/ClubChatRoom/ChatRoom 삭제

## 💻 환경 정보
- OS / 브라우저: macOS
- 요청 경로(URL): POST /api/clubs, DELETE /api/clubs/{clubId} 등 연관 생성/삭제 엔드포인트
- 기타 환경 정보:
